src/Analysis/GWASV2/Shared/ProgressBar/ProgressBar.test.jsx

```diff
@@ -0,0 +1,135 @@
+describe('Test that active step class renders with active class when current is two', () => {
+  const wrapper = mount(<ProgressBar current={2} />);
+  it('should render third step with active class when current is two', () => {
+    testElementClass(wrapper, 3, 'ant-steps-item-active');
+  });
+});
+
+describe('Test that active step class renders with active class when current is three', () => {
+  const wrapper = mount(<ProgressBar current={3} />);
+  it('should render fourth step with active class when current is three', () => {
+    testElementClass(wrapper, 4, 'ant-steps-item-active');
+  });
+});
+
+/* Test Dynamic Text for Steps */
+describe('Test that each step renders with correct text when current is zero', () => {
+  const wrapper = render(<ProgressBar current={0} />);
+  it('should render first step', () => {
+    testElementText(wrapper, 1, '1Select Study Population');
+  });
+  it('should render second step', () => {
+    testElementText(wrapper, 2, '2Select Outcome Phenotype');
+  });
+  it('should render third step', () => {
+    testElementText(wrapper, 3, '3Select Covariate Phenotype');
+  });
+  it('should render fourth step', () => {
+    testElementText(wrapper, 4, '4Configure GWAS');
+  });
+});
+
+describe('Test that each step renders with correct text when current is one', () => {
+  const wrapper = render(<ProgressBar current={1} />);
+  it('should render first step', () => {
+    testElementText(wrapper, 1, '1Edit Study Population');
+  });
+  it('should render second step', () => {
+    testElementText(wrapper, 2, '2Select Outcome Phenotype');
+  });
+  it('should render third step', () => {
+    testElementText(wrapper, 3, '3Select Covariate Phenotype');
+  });
+  it('should render fourth step', () => {
+    testElementText(wrapper, 4, '4Configure GWAS');
+  });
+});
+
+describe('Test that each step renders with correct text when current is two', () => {
+  const wrapper = render(<ProgressBar current={2} />);
+  it('should render first step', () => {
+    testElementText(wrapper, 1, '1Edit Study Population');
+  });
+  it('should render second step', () => {
+    testElementText(wrapper, 2, '2Edit Outcome Phenotype');
+  });
+  it('should render third step', () => {
+    testElementText(wrapper, 3, '3Select Covariate Phenotype');
+  });
+  it('should render fourth step', () => {
+    testElementText(wrapper, 4, '4Configure GWAS');
+  });
+});
+
+describe('Test that each step renders with correct text when current is three', () => {
+  const wrapper = render(<ProgressBar current={3} />);
+  it('should render first step', () => {
+    testElementText(wrapper, 1, '1Edit Study Population');
+  });
+  it('should render second step', () => {
+    testElementText(wrapper, 2, '2Edit Outcome Phenotype');
+  });
+  it('should render third step', () => {
+    testElementText(wrapper, 3, '3Edit Covariate Phenotype');
+  });
+  it('should render fourth step', () => {
+    testElementText(wrapper, 4, '4Configure GWAS');
+  });
+});
```

Comment:
thanks for adding tests 👍 

I would like to propose a few simplifications to enhance the maintainability without loosing much:
- `should render ... step with active class when current is ...` -> implement a loop of 4?
- "Test Dynamic Text for Step" -> this could also be a single test with a loop instead of 4 tests. Also do not test against the whole text but just check if the step starts with "Edit". In this way, the test does not break if we decide to make small changes to the title.

Reply:
Interesting idea! I did this and got it to work. 

A concern of mine is that this approach can lead to more difficult to interpret tests. For example: if something breaks it might be easier for a developer to quickly interpret explicitly stated test code compared with the same logic implemented with loops.

Reply:
I agree that test readability is also super important. I think the tests are still readable.